TTWWW-477: Prevalence Map "cite the map" functionality

mattga11ant opened the pull request:
Referencing this Jira ticket: https://simonsfoundation.atlassian.net/browse/TTWWW-477

Noting that the DOI link that is shown in Figma is not valid. That is the value I have in place for now though. I tested using a valid DOI link from the TT site, and verified it worked as expected. I could not find a DOI link on the TT site that had any citations though, so all I was able to confirm was that 0 is returned successfully for a link that has 0 citations.

I chose to hide the tooltip that says "Copied to clipboard" after 3 seconds. It made sense to do that, and not have that tooltip stay open or require clicking again to hide it.

On hover, I have the "CITE THE MAP (0)" text changing to the red color, but was not sure if we wanted to make the SVG and the URL text also change color.

Finally I added a placeholder for the "Calculate Mean" button so that I could style that entire container with all three buttons correctly.

- [x] pull this branch
- [x] compile CSS and JS updates with npm run dev
- [x] confirm that the style of the Cite The Map button matches the design in Figma
- [x] confirm that hovering the button, it changes to red
- [x] confirm that clicking the button shows the tooltip, which hides after 3 seconds
- [x] confirm that clicking the button copies the link to the clipboard
- [ ] TODO: confirm that the citation count is correct by using a known DOI link, or by creating one for this site and testing

yjfnyc reviewed (changes requested):
Worked as described, and code looked good. But I saw the layout at 1280px was like the following
<img width="1425" alt="Screenshot 2025-02-15 at 10 54 39 PM" src="https://github.com/user-attachments/assets/6571f464-b277-474c-9d05-64be26d22ca1" />
Matt, can you please take a further look on your end to see if that is also the case? The DOI url is updated in the ticket comment [here](https://simonsfoundation.atlassian.net/browse/TTWWW-477?focusedCommentId=109124)

mattga11ant commented:
> Worked as described, and code looked good. But I saw the layout at 1280px was like the following <img alt="Screenshot 2025-02-15 at 10 54 39 PM" width="1425" src="https://private-user-images.githubusercontent.com/9726273/413587391-6571f464-b277-474c-9d05-64be26d22ca1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MTM4MzgsIm5iZiI6MTczOTgxMzUzOCwicGF0aCI6Ii85NzI2MjczLzQxMzU4NzM5MS02NTcxZjQ2NC1iMjc3LTQ3NGMtOWQwNS02NGJlMjZkMjJjYTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTdUMTczMjE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWI4YzRjZWNiNmQwNzgzNzFkYzc0MmUzNWNlNjIwNTU0ODNlN2Q2Yjk4ZTA1NGYyNWI2ODlmMDFkZjY4ZGYxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Gg3HPTp9D1km17esEUcK_1VbSUy925dpffCTJQsaZ8g"> Matt, can you please take a further look on your end to see if that is also the case? The DOI url is updated in the ticket comment [here](https://simonsfoundation.atlassian.net/browse/TTWWW-477?focusedCommentId=109124)

Good catch on the design at 1280px. I have updated this to use absolute positioning, which allows the slight overlap shown in the design. When you continue to go smaller than 1280px, the buttons will overlap the filters. Given that we are not accounting for smaller than 1280px, should I do anything else to address that overlap? Like hiding that container under 1280px?

I also have the correct DOI link now added.

yjfnyc commented:
@mattga11ant Thanks for the update, but I see overlapping as shown on the following screenshot
<img width="1425" alt="Screenshot 2025-02-17 at 1 47 16 PM" src="https://github.com/user-attachments/assets/7b3f9352-19aa-47c2-93b4-4b3e110dd437" />

mattga11ant commented:
> @mattga11ant Thanks for the update, but I see overlapping as shown on the following screenshot <img alt="Screenshot 2025-02-17 at 1 47 16 PM" width="1425" src="https://private-user-images.githubusercontent.com/9726273/413983089-7b3f9352-19aa-47c2-93b4-4b3e110dd437.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MTg3OTIsIm5iZiI6MTczOTgxODQ5MiwicGF0aCI6Ii85NzI2MjczLzQxMzk4MzA4OS03YjNmOTM1Mi0xOWFhLTQ3YzItOTNiNC00YjNlMTEwZGQ0MzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTdUMTg1NDUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmQ2ZjQwY2I2NzA5ZGM3YjQ5YzIxNmIzNzkxN2I2OTAzZGQ4NmJjZmM1NDcxNmMxODUyNWYxMjRjNTE2OTYzMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Qu3sEUcmjmrT5aXHNrmvU05HIgJxDOCPMg0Iifjlot0">

@yjfnyc In your screenshot, it looks like your browser may be zoomed in (I see the magnifying glass to the right of your address bar). This is what I see at 1280px wide locally, which matches what is shown in Figma.

<img width="1257" alt="Screenshot 2025-02-17 at 11 54 43 AM" src="https://github.com/user-attachments/assets/f65ce850-43ff-4c7a-aa9c-11fd6908ca92" />

yjfnyc commented:
@mattga11ant thanks for pointing it out. That is exactly the case. I also tested on firefox, it matched.

yjfnyc reviewed (approved):
looks good to me.
